src/Features/CSharp/Portable/AddBraces/CSharpAddBracesCodeFixProvider.cs

```diff
@@ -1,17 +1,37 @@
-﻿using System.Composition;
+﻿using System;
+using System.Collections.Immutable;
+using System.Composition;
 using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.AddBraces;
+using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeFixes;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
 {
     [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddBraces), Shared]
     [ExtensionOrder(After = PredefinedCodeFixProviderNames.AddAwait)]
-    internal class CSharpAddBracesCodeFixProvider : AbstractAddBracesCodeFixProvider
+    internal class CSharpAddBracesCodeFixProvider : CodeFixProvider
     {
-        protected override async Task<Document> AddBracesAsync(CodeFixContext context)
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(IDEDiagnosticIds.AddBracesDiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    FeaturesResources.AddBraces,
+                    async c => await AddBracesAsync(context).ConfigureAwait(false)),
```

Comment:
you can just write this as: ```c => AddBracesAsync(context, c)```.  Note that want to pass along this cancellation token and *not* use the one from context.